docs/Displaying a User Interface/Alerts/index.md

```diff
@@ -1,135 +1,217 @@
 # Alerts
 An alert is a pop-up window that can show a short message with optional buttons. When an alert is activated, it will abort any SDL operation that is in-progress, except the already-in-progress alert. If an alert is issued while another alert is still in progress, the newest alert will simply be ignored.
 
+Depending the platform, an alert can have up to three lines of text, a progress indicator (e.g. a spinning wheel or hourglass), and up to four soft buttons.
+
 !!! NOTE
 The alert will persist on the screen until the timeout has elapsed, or the user dismisses the alert by selecting a button. There is no way to dismiss the alert programmatically other than to set the timeout length.
 !!!
 
-## Alert UI
-Depending the platform, an alert can have up to three lines of text, a progress indicator (e.g. a spinning wheel or hourglass), and up to four soft buttons.
+## Creating an Alert
+### Alert With No Soft Buttons
+!!! NOTE
+If no soft buttons are added to an alert some OEMs may add a default "cancel" or "close" button.
+!!!
 
-### Alert HMI
-###### Generic HMI
+#### Alert HMI
 ![Generic - Alert](assets/Generic_alert.png)
 
-###### Ford HMI
-![SYNC 3 - Alert](assets/SYNC3_alert.jpg)
+@![iOS]
+#####  Objective-C
+```objc
+SDLAlert *alert =  [SDLAlert alloc] initWithAlertText1:@"<#Line 1#>" alertText2:@"<#Line 2#>" alertText3:@"<#Line 3#>"];
+```
 
-## Alert Text-To-Speech
-The alert can also be formatted to speak a prompt when the alert appears on the screen. Do this by setting the `ttsChunks` parameter. To play the alert tone before the text-to-speech is spoken, set `playTone` to `true`.
+##### Swift
+```swift
+let alert = SDLAlert(alertText1: "<#Line 1#>", alertText2: "<#Line 2#>", alertText3: "<#Line 3#>")
+```
+!@
+
+@![android,javaSE,javaEE]
+` TODO - code example `
+!@
+
+### Alert With Soft Buttons
+
+#### Alert HMI
+![Generic - Alert](assets/Generic_alert_buttons.png)
 
-## Create an Alert
 @![iOS]
 ##### Objective-C
```

Comment:
Move these code examples down. See Nicole's issue for the general structure. These would be good for button examples at the bottom.

Reply:
I think the order should be in the best order now after the other changes